backend/tests/integration/test_file_upload_validation.py

```
@@ -0,0 +1,505 @@
+
+        # Verify reasonable length (should be <= 255 chars)
+        assert (
+            len(filename) <= 255
+        ), f"Sanitized filename too long: {len(filename)} chars"
+
+        # Verify filename is not empty and not just dots/spaces
+        assert filename.strip(" ."), "Sanitized filename is empty or only dots/spaces"
+
+    def test_end_to_end_filename_security(self):
+        """Test that malicious filenames don't create dangerous files on filesystem"""
+        import tempfile
```

Ruff (0.11.9)

Redefinition of unused `tempfile` from line 7

Remove definition: `tempfile`

(F811)